server/src/main/java/org/elasticsearch/script/ScriptContextStats.java

```diff
@@ -43,6 +56,53 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVLong(compilations);
         out.writeVLong(cacheEvictions);
         out.writeVLong(compilationLimitTriggered);
+        if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+            out.writeOptionalWriteable(compilationsHistory);
+            out.writeOptionalWriteable(cacheEvictionsHistory);
+        }
+    }
+
+    public static class TimeSeries implements Writeable, ToXContentFragment {
+        public final long five;
+        public final long fifteen;
+        public final long day;
+
+        public TimeSeries() {
+            this.five = 0;
+            this.fifteen = 0;
+            this.day = 0;
+        }
+
+        public TimeSeries(long five, long fifteen, long day) {
+            this.five = five;
```

Comment:
We should add assertions here to ensure the data is not garbage. If five != 0, then fifteen must also be non zero, and if fifteen is non zero, day must also be non zero.

Reply:
Great suggestion, caught some bugs when I added the tests you suggested above.